data_safe_haven/serialisers/azure_serialisable_model.py

```diff
@@ -14,14 +14,16 @@ class AzureSerialisableModel(YAMLSerialisableModel):
     """Base class for configuration that can be written to Azure storage"""
 
     config_type: ClassVar[str] = "AzureSerialisableModel"
-    filename: ClassVar[str] = "config.yaml"
+    default_filename: ClassVar[str] = "config.yaml"
 
     @classmethod
-    def from_remote(cls: type[T], context: ContextBase) -> T:
+    def from_remote(
+        cls: type[T], context: ContextBase, *, filename: str | None = None
+    ) -> T:
         """Construct an AzureSerialisableModel from a YAML file in Azure storage."""
         azure_api = AzureApi(subscription_name=context.subscription_name)
         config_yaml = azure_api.download_blob(
-            cls.filename,
+            filename or cls.default_filename,
```

Comment:
Could we avoid the extra argument here by just using `cls.filename` and use the `DSHSREConfig.filename` property?

Reply:
No, because you need to be able to load as a class-level call to `SREConfig.from_remote` and `SREConfig.filename` doesn't exist until you've created an object.